diff --git a/node_modules/@chadify/ffmpeg-kit-react-native/android/build.gradle b/node_modules/@chadify/ffmpeg-kit-react-native/android/build.gradle
index 2909280..10ae93d 100644
--- a/node_modules/@chadify/ffmpeg-kit-react-native/android/build.gradle
+++ b/node_modules/@chadify/ffmpeg-kit-react-native/android/build.gradle
@@ -122,8 +122,14 @@ repositories {
     )
   }
 }
+repositories {
+  flatDir {
+    dirs project(':app').file('libs')
+  }
+}
 
 dependencies {
   api 'com.facebook.react:react-native:+'
-  implementation 'com.arthenica:ffmpeg-kit-' + safePackageName(safeExtGet('ffmpegKitPackage', 'https')) + ':' + safePackageVersion(safeExtGet('ffmpegKitPackage', 'https'))
+  // Use local AAR instead of broken remote dependency
+  implementation(name: 'ffmpeg-kit-full-gpl', ext: 'aar')
 }
